### ISSUE TITLE
How to consume an Amplify backend from web and react native (2 apps)?

### ISSUE THREAD

**Issue by rawadrifai** [aws-amplify/amplify-cli#1716]
** Which Category is your question related to? **
Multiple frontends for the same backend.

** What AWS Services are you utilizing? **
The whole amplify stack.

** Provide additional details e.g. code snippets **
I created a react native app that consumes the backend created in a react project hooked with amplify. I simply copy the aws-exports.js and the mutations/queries I need, and I use them from the other app. Is this the best practice?

**Comment by jkeys-ecg-nmsu** [comment]
+1 but scaled to `n` frontend applications.

**Comment by mikeparisstuff** [comment]
@rawadrifai You can use `amplify add codegen --apiId <api-id>` from another app that is not directly tied to the Amplify project.

**Comment by jkeys-ecg-nmsu** [comment]
@mikeparisstuff just to clarify, the recommended flow is to create your "primary" project with your AppSync backend and develop your "primary" client on this project, then create as many secondary projects as you have applications and manually add the primary API back to them?

If that's the case I feel like everyone who wonders about this architectural question is a +1 on my existing FR (#898) about bifurcating the concept of template and concrete environments. We should be able to export a template environment that references hardcoded resources as needed if this is the intended workflow.

**Comment by jkeys-ecg-nmsu** [comment]
@mikeparisstuff please track this concern. We have been using this solution in a different way: just importing a renamed "aws-exports-primary.js" and combining objects in code. If this is the preferred method it should be documented as the recommend workflow.

**Comment by undefobj** [comment]
I think there is some confusion here, sharing the same backend between multiple frontends is outlined here: https://aws-amplify.github.io/docs/cli/init#amplify-configure-project

> Note: When sharing your backend infrastructure between multiple frontend projects or between different machines, you will need to first manually copy the amplify/ directory from your current project to the other project’s root directory and then run the amplify configure project command.

**Comment by rawadrifai** [comment]
@undefobj It was working for me with just aws-exports.js and you run Amplify.configure(..) and it worked out well.

Today I decided to remove my backend and reinstall it. The rn native app won't work anymore, specifically I can't log in with withAuthenticator.. it says "User does not exist" even when I try to sign up.
p.s. I tried changing "aws_user_pools_web_client_id" in aws-exports.js to match the second app client id created in Cognito under App Clients.
I'm stuck.
![image](https://user-images.githubusercontent.com/24530625/60072980-07d2cf00-96d4-11e9-84c9-fc97606626ed.png)
